components/exercises/chapter-4-oscar/Utils.test.tsx

```diff
@@ -80,3 +88,190 @@ describe('reverseArrayInPlace', () => {
     expect(arr2).toEqual(['e', 'd', 'c', 'b', 'a']);
   });
 });
+
+describe('arrayToList', () => {
+  it('should convert an array of numbers into a linked list', () => {
+    const arr = [1, 2, 3];
+    const expectedList = {
+      value: 1,
+      rest: {
+        value: 2,
+        rest: {
+          value: 3,
+          rest: null,
+        },
+      },
+    };
+    expect(arrayToList(arr)).toEqual(expectedList);
+  });
+  it('should convert an array of strings into a linked list', () => {
+    const arr = ['a', 'b', 'c'];
+    const expectedList = {
+      value: 'a',
+      rest: {
+        value: 'b',
+        rest: {
+          value: 'c',
+          rest: null,
+        },
+      },
+    };
+    expect(arrayToList(arr)).toEqual(expectedList);
+  });
+
+  it('should return null for an empty array', () => {
+    expect(arrayToList([])).toBeNull();
+  });
+});
+
+describe('listToArray', () => {
+  it('should convert a linked list to an array', () => {
+    const list = {
+      value: 1,
+      rest: {
+        value: 2,
+        rest: {
+          value: 3,
+          rest: null,
+        },
+      },
+    };
+    const expectedArray = [1, 2, 3];
+    expect(listToArray(list)).toEqual(expectedArray);
+  });
```

Comment:
Can we add a test for array length different than 3?

Reply:
Test added